spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -73,7 +74,10 @@ public ConsumerDestination provisionConsumerDestination(String topicName, String
 
 		// topicName may be either the short or fully-qualified version.
 		String topicShortName = TopicName.isParsableFrom(topicName) ? TopicName.parse(topicName).getTopic() : topicName;
-		Optional<Topic> topic = ensureTopicExists(topicName, properties.getExtension().isAutoCreateResources());
+		Topic topic = ensureTopicExists(topicName, properties.getExtension().isAutoCreateResources());
+
+		PubSubConsumerProperties.DeadLetterPolicy deadLetterPolicy = properties.getExtension().getDeadLetterPolicy();
+		boolean autoCreate = properties.getExtension().isAutoCreateResources();
```

Comment:
Might want to move it to above line 77, and just use that variable. 
More broadly, perhaps none of this logic needs to run if `autoCreateResources` is off?